pkg/admission/admission.go

```diff
@@ -3,19 +3,32 @@ package admission
 import (
 	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 	"reflect"
 	"time"
 
 	"github.com/golang/glog"
 	"github.com/mattbaird/jsonpatch"
 
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
 )
 
-func New(listenAddress string) *http.Server {
+type admissionData struct {
+	coreClient kubernetes.Interface
+}
+
+var jsonPatch = admissionv1beta1.PatchTypeJSONPatch
+
+func New(listenAddress string, coreClient kubernetes.Interface) *http.Server {
 	m := http.NewServeMux()
-	m.HandleFunc("/machinedeployments", handleFuncFactory(mutateMachineDeployments))
+	ad := &admissionData{coreClient: coreClient}
+	m.HandleFunc("/machinedeployments", handleFuncFactory(ad.mutateMachineDeployments))
+	m.HandleFunc("/machines", handleFuncFactory(ad.mutateMachines))
```

Comment:
Should we also cover `MachineSet`s ? (Not in this PR of course)

Reply:
Yes, will add that in a follow-up